### PR TITLE
Fix #1239, scrub include header guards

### LINKED ISSUE #1239: Header guard update to match osal pattern
**Is your feature request related to a problem? Please describe.**
Current header guards utilized reserved leading `_`, likely also inconsistent

Helpful if it also catches endif comment (typically comments the matching define)

**Describe the solution you'd like**
Apply OSAL pattern

**Describe alternatives you've considered**
None

**Additional context**
None

**Requester Info**
Jacob Hageman - NASA/GSFC

### PR DESCRIPTION
**Pull request by jphickey**
**Describe the contribution**
All C header files should have a header guard matching the file name of the header, in ALL_CAPS, with no extra leading/trailing
underscores.

This also takes a first pass at converting the file-scope block comments to a doxygen format to include a summary of the file.

Fixes #1239

**Testing performed**
Build and sanity check CFE, build and run all unit tests
Build documentation and confirm the content of "Detailed Description" on the files contains the info from `@file` doxygen blocks.

**Expected behavior changes**
No impact to behavior.

**System(s) tested on**
Ubuntu 20.04

**Contributor Info - All information REQUIRED for consideration of pull request**
Joseph Hickey, Vantage Systems, Inc.

### CODE REVIEW

File: cmake/sample_defs/cpu1_platform_cfg.h [hunk 1]
```diff
@@ -18,23 +18,24 @@
...
-** Author:   R.McGraw/SSI
-**
-******************************************************************************/
-
-#ifndef _cfe_platform_cfg_
-#define _cfe_platform_cfg_
+/**
+ * @file
+ *
+ *
```
Comment (skliper) [inline cmake/sample_defs/cpu1_platform_cfg.h:24]: Extra line?  Looks like it's in most (all?) of the replaced file documentation

Comment (skliper) [reply cmake/sample_defs/cpu1_platform_cfg.h:24]: maybe replace all `/n */n */n` with `/n */n' and (nitpick) then all '/n */n *///n` with '/n *///n` (remove extra empty comment line at the end f I got the escapes right)

Comment (jphickey) [reply cmake/sample_defs/cpu1_platform_cfg.h:24]: Yeah I noticed that too - the script I used didn't quite get it all right.  I will run another pass to get multiple blank lines in comments.  actually a simple "uniq" might do it.

Comment (jphickey) [reply cmake/sample_defs/cpu1_platform_cfg.h]: Addressed in commit 0780d12

**Pull request merged**